### PR TITLE
Blog onboarding: Address "Already own a domain"

### PR DESCRIPTION
**Pull request by lupus2k**
Fixes https://github.com/Automattic/dotcom-forge/issues/2329

## Proposed Changes

On `Choose a domain` step, we have the option to use your own domain (bottom of the page). We want to make this option works, add the domain mapping to the cart, and update the design.

https://user-images.githubusercontent.com/402286/236369437-f1ac9b31-643f-4ee9-a081-dd728ad0f239.mov

## Known issues
- [x] The W icon is not showing on `Use a domain I own` step
- [x] The subtitle copy is not correct on `Use a domain I own` step.
- [x] The domain mapping is not added to the cart.
- [ ] It was not tested with transferable domains.
- [ ] The design is not updated on the domains step.

## Testing Instructions

* Using a new user OR a user without any site, start the flow with [/setup/start-writing](http://calypso.localhost:3000/setup/start-writing).
* On the `Choose a domain` step, select `Already own a domain? You can use it as your site’s address` at the bottom of the page.
* Follow the steps.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/use-my-domain/index.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+				stepName="useMyDomain"
+				shouldHideNavButtons={ isStartWritingFlow }
+				goBack={ goBack }
+				goNext={ goNext }
+				isHorizontalLayout={ false }
+				isWideLayout={ true }
+				isLargeSkipLayout={ false }
+				stepContent={ getStepContent() }
+				recordTracksEvent={ recordTracksEvent }
+				formattedHeader={ getFormattedHeader() }
```
Comment (DustyReagan) [inline client/landing/stepper/declarative-flow/internals/steps-repository/use-my-domain/index.tsx]: I'm wondering if we passed `formattedHeader={ undefined }` here, if we could simplify a couple things by using the header in `UseMyDomainComponent`. Might be able to touch one less file and create one less function. How do you see it?

**Review by DustyReagan — APPROVED** [review]
Works great! Nice job! 👍 I left a comment for consideration.

**Review by paulopmt1 — APPROVED** [review]
Good job @lupus2k! At the end, I could buy a connection to my domain: 

image

File: client/landing/stepper/declarative-flow/internals/steps-repository/use-my-domain/index.tsx (hunk 1 above)
Comment (DustyReagan) [reply client/landing/stepper/declarative-flow/internals/steps-repository/use-my-domain/index.tsx]: Discussed here p1683721966816739-slack-CKZHG0QCR and implemented. ☺️

**Pull request merged**